core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.ts

```diff
@@ -13,543 +13,641 @@
-            });
-        }, function() {
-          // Note to developers:
-          // This callback is triggered when the Cancel button is clicked.
-          // No further action is needed.
-          $rootScope.$applyAsync();
-        });
+import { Rubric } from 'domain/skill/rubric.model';
+import { SkillBackendApiService } from 'domain/skill/skill-backend-api.service';
+import { MisconceptionSkillMap } from 'domain/skill/MisconceptionObjectFactory';
+import { Question, QuestionBackendDict, QuestionObjectFactory } from 'domain/question/QuestionObjectFactory';
+import { ActiveContributionDict, TranslationSuggestionReviewModalComponent } from '../modal-templates/translation-suggestion-review-modal.component';
+import { ContributorDashboardConstants } from 'pages/contributor-dashboard-page/contributor-dashboard-page.constants';
+import { QuestionSuggestionReviewModalComponent } from '../modal-templates/question-suggestion-review-modal.component';
+import { TranslationTopicService } from 'pages/exploration-editor-page/translation-tab/services/translation-topic.service';
+import { FormatRtePreviewPipe } from 'filters/format-rte-preview.pipe';
+import { UserService } from 'services/user.service';
+import { AlertsService } from 'services/alerts.service';
+import { ContextService } from 'services/context.service';
+import { ContributionAndReviewService } from '../services/contribution-and-review.service';
+import { ContributionOpportunitiesService } from '../services/contribution-opportunities.service';
+
+export interface Suggestion {
+  change: {
+    skill_id?: string;
+    content_html: string;
+    translation_html: string | string[];
+    question_dict?: QuestionBackendDict;
+    skill_difficulty?: string[];
+  };
+  status: string;
+  suggestion_type: string;
+  target_id: string;
+  suggestion_id: string;
+  author_name?: string;
+}
+
+export interface QuestionContributionsSummaryList {
+  id: string;
+  heading: string;
+  subheading: string;
+  labelText: string;
+  labelColor: string;
+  actionButtonTitle: string;
+}
+
+export interface TranslationContributionsSummaryList {
+  id: string;
+  heading: string;
+  subheading: string;
+  labelText: string;
+  labelColor: string;
+  actionButtonTitle: string;
+}
+
+export interface OpportunitiesDictsObject {
+  opportunitiesDicts: unknown;
+  more: unknown;
+}
+
+export interface ContributionDetails {
+  skill_description: string;
+  skill_rubrics: Rubric[];
+  chapter_title: string;
+  story_title: string;
+  topic_name: string;
+}
+
+export interface ContributionTabs {
+  suggestionType: string;
+  text: string;
+  enabled: boolean;
+}
+
+export interface SuggestionIdToSuggestions {
+  [key: string]: {
+    suggestion: Suggestion;
+    details: {
+      skill_description: string;
+      topic_name: string;
+      chapter_title: string;
+      story_title: string;
+    };
+  };
+}
+
+export interface SUGGESTION_LABELS {
+  [key: string]: {
+    text: string;
+    color: string;
+  };
+}
```

Comment:
Do we need a type for this? Can we just declare as a `const` constant below instead?

Reply:
removed